x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrIndexLifecycleIT.java

```diff
@@ -0,0 +1,171 @@
+        final String followerIndex = "follower";
+        final String policyName = "unfollow_only_policy";
+        final int numberOfShards = randomIntBetween(1, 4);
+        final Map<String, String> indexSettings = new HashMap<>();
+        indexSettings.put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), "true");
+        indexSettings.put(LifecycleSettings.LIFECYCLE_NAME_SETTING.getKey(), policyName);
+        final String leaderIndexSettings =
+            getIndexSettings(numberOfShards, 0, indexSettings);
+        assertAcked(leaderClient().admin().indices().prepareCreate(leaderIndex).setSource(leaderIndexSettings, XContentType.JSON).get());
+
+        // Put a policy on the follower
+        Map<String, LifecycleAction> hotActions = new HashMap<>();
+        hotActions.put(org.elasticsearch.xpack.core.indexlifecycle.UnfollowAction.NAME,
+            new org.elasticsearch.xpack.core.indexlifecycle.UnfollowAction());
+        Map<String, Phase> phases = new HashMap<>();
+        phases.put("hot", new Phase("hot", TimeValue.ZERO, hotActions));
+        LifecyclePolicy policy = new LifecyclePolicy(policyName, phases);
+
+        PutLifecycleAction.Request putLifecycleRequest = new PutLifecycleAction.Request(policy);
+        assertAcked(followerClient().execute(PutLifecycleAction.INSTANCE, putLifecycleRequest).get());
+
+        // Set up the follower
+        final PutFollowAction.Request followRequest = putFollow(leaderIndex, followerIndex);
+        followerClient().execute(PutFollowAction.INSTANCE, followRequest).get();
+        ensureFollowerGreen(true, followerIndex);
+
+        // Pause ILM so that this policy doesn't proceed until we want it to
+        assertAcked(followerClient().execute(StopILMAction.INSTANCE, new StopILMRequest()).get());
+
+        // Set indexing complete and wait for it to be replicated
+        assertAcked(leaderClient().admin().indices().prepareUpdateSettings(leaderIndex)
+            .setSettings(Settings.builder().put(LifecycleSettings.LIFECYCLE_INDEXING_COMPLETE_SETTING.getKey(), "true"))
+            .get());
+        assertBusy(() -> assertEquals("true",
+            followerClient().admin().indices()
+                .getIndex(new GetIndexRequest().indices(followerIndex)).get()
+                .getSetting(followerIndex, LifecycleSettings.LIFECYCLE_INDEXING_COMPLETE_SETTING.getKey())));
+
+        // Disrupt requests to remove retention leases for these random shards
+        final Set<Integer> shardIds =
+            new HashSet<>(randomSubsetOf(
+                randomIntBetween(1, numberOfShards),
+                IntStream.range(0, numberOfShards).boxed().collect(Collectors.toSet())));
+
+        final ClusterStateResponse followerClusterState = followerClient().admin().cluster().prepareState().clear().setNodes(true).get();
+        try {
+            for (final ObjectCursor<DiscoveryNode> senderNode : followerClusterState.getState().nodes().getNodes().values()) {
```

Comment:
Alternatively this test can also set `cluster.remote.leader_cluster.seeds` setting to `null` via cluster update settings api, which disables the follower cluster to communicate to the leader cluster.  (this test would then also need to extend overwrite `configureRemoteClusterViaNodeSettings()` and return false).

I think this is a bit cleaner, as it it requires fewer lines of code and doesn't involve interacting with mock transport service. But since this is a common practice in other tests, I think it is okay for this test too. Just wanted to mention the alternative. Also the upside of the current approach, is that it is a more realistic simulation of when the leader cluster isn't reachable from the follower cluster in production.

Reply:
Thanks - I'll try this. If that's the case, then this might not need to be a `CcrIntegTestCase` and we could roll it into the existing REST-based test cases in the ILM module, rather than having to add a dependency on CCR.

I didn't like that this test was in CCR, rather than ILM, but I wanted to get eyes on the code, and it can always be moved.  I'll look into this and update this PR moving this test into the ILM module one way or another.